packages/core/saved-objects/core-saved-objects-server-internal/src/apply_type_defaults.test.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { SavedObjectsType } from '@kbn/core-saved-objects-server';
+import { globalSwitchToModelVersionAt } from '@kbn/core-saved-objects-base-server-internal';
+import { applyTypeDefaults } from './apply_type_defaults';
+
+const createType = (parts: Partial<SavedObjectsType> = {}): SavedObjectsType => ({
+  name: 'test',
+  namespaceType: 'single',
+  hidden: false,
+  mappings: { properties: {} },
+  ...parts,
+});
+
+describe('applyTypeDefaults', () => {
+  describe('switchToModelVersionAt', () => {
+    it(`keeps the type's switchToModelVersionAt if lesser than the global version`, () => {
+      const type = createType({
+        switchToModelVersionAt: '8.4.0',
+      });
+
+      const result = applyTypeDefaults(type);
+      expect(result.switchToModelVersionAt).toEqual('8.4.0');
+    });
+
+    it(`sets switchToModelVersionAt to the global version if type version is greater than the global version`, () => {
+      const type = createType({
+        switchToModelVersionAt: '9.2.0',
+      });
+
+      const result = applyTypeDefaults(type);
+      expect(result.switchToModelVersionAt).toEqual(globalSwitchToModelVersionAt);
+    });
```

Comment:
should this be allowed as we mentioned we needed a workaround to give teams some extra time?

Reply:
We could use a `delaySwitchToModelVersionAt` and override the default or not apply the forced switch to those. That way, we need to agree on which types we're more lenient with based on the type-owner's justification.
Given the timing, we do need to force our hand a bit here.

Reply:
Let's start by not :) and figure out if we need it then

Reply:
IMO, there is no need for an additional property. The way I understood our conversation was:

If not provided, let's set `8.10.0` as the default switchover deadline.
Teams can override that default by either anticipating or postponing the switch. If postponing, we'd discuss it in the PR review.

Basically `function registerType({ switchToModelVersionAt = '8.10.0', ...otherOptions }) {...}`.

Given the short timeframe we are providing, I'm almost certain that _some_ teams will need to postpone it.

But I don't think this is a blocker, so let's keep it as it is, and we can soften the validation in a follow-up PR if needed.